From 0d0b62575f2a52710d1fc3d59ed375cb3a50701a Mon Sep 17 00:00:00 2001
From: Kevin Hellemun <17928966+OGKevin@users.noreply.github.com>
Date: Mon, 21 Sep 2026 12:58:03 +0200
Subject: [PATCH] fix(gcc 4.9.4): compilation errors

The current build for v14.5.0 fails to compile on gcc 4.9.4 which is
used on embedded Kobo systems.

Ref: https://github.com/OGKevin/cadmus/pull/1005

Change-Id: 7a4bec5b5418bd0921b273d8d4a1e79b
Change-Id-Short: spvolnuouvyr
Upstream: https://github.com/harfbuzz/harfbuzz/commit/0d0b62575f2a52710d1fc3d59ed375cb3a50701a
Signed-off-by: Raphaël Mélotte <raphael.melotte@mind.be>
---
 src/graph/graph.hh | 2 +-
 src/test-result.cc | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/graph/graph.hh b/src/graph/graph.hh
index 0399f7f3e1f..e0e733e7e61 100644
--- a/src/graph/graph.hh
+++ b/src/graph/graph.hh
@@ -635,7 +635,7 @@ struct graph_t
     }
     TRY(r);
 
-    return g;
+    return Ok (std::move (g));
   }
 
   graph_t (const graph_t&) = delete;
diff --git a/src/test-result.cc b/src/test-result.cc
index 9d487d4083e..b7b93a4cbe7 100644
--- a/src/test-result.cc
+++ b/src/test-result.cc
@@ -433,7 +433,7 @@ int copy_move_counter_t::move_count = 0;
 static result<copy_move_counter_t> return_tracked_from_local ()
 {
   copy_move_counter_t out (42);
-  return out;
+  return Ok (std::move (out));
 }
 
 static result<copy_move_counter_t> return_tracked_from_local_with_tag ()
@@ -445,7 +445,7 @@ static result<copy_move_counter_t> return_tracked_from_local_with_tag ()
 static hb_result_t<int, copy_move_counter_t> return_tracked_error_from_local ()
 {
   copy_move_counter_t out (42);
-  return out;
+  return Err (std::move (out));
 }
 
 static hb_result_t<int, copy_move_counter_t> return_tracked_error_from_local_with_tag ()
@@ -457,7 +457,7 @@ static hb_result_t<int, copy_move_counter_t> return_tracked_error_from_local_wit
 static result<move_only_val_t> return_move_only_from_local ()
 {
   move_only_val_t out (42);
-  return out;
+  return Ok (std::move (out));
 }
 
 static void test_return_moves_from_local ()
